Source/WebCore/html/FileInputType.cpp

```diff
@@ -371,7 +371,7 @@ void FileInputType::setFiles(RefPtr<FileList>&& files, RequestIcon shouldRequest
         pathsChanged = true;
     else {
         for (unsigned i = 0; i < length; ++i) {
-            if (files->file(i).path() != m_fileList->file(i).path()) {
+            if (files->file(i).path() != m_fileList->file(i).path() || files->file(i).lastModifiedOverride() != m_fileList->file(i).lastModifiedOverride()) {
```

Comment:
Doesn't this mean that we'd consider two files of the same name & same last modified date as identical?
Why not use the unique identifier of the filesystem instead?
This is stat.st_ino in POSIX, and [FileId in FILE_ID_BOTH_DIR_INFO](https://learn.microsoft.com/en-us/windows/win32/api/winbase/ns-winbase-file_id_both_dir_info) for Windows. [G_FILE_ATTRIBUTE_UNIX_INODE](https://www.freedesktop.org/software/gstreamer-sdk/data/docs/2012.5/gio/GFileInfo.html#G-FILE-ATTRIBUTE-STANDARD-NAME:CAPS) for glib subsystem.

Reply:
I didn't know those existed! How would I use the Windows and glib functions? (I haven't written many platform specific functions yet)

Reply:
You use GetFileInformationByHandleEx to get the information. See `fileSize(PlatformFileHandle fileHandle)` in FileSystemWin.cpp for an example of calling a similar function: GetFileInformationByHandle